src/OrleansCodeGenerator/SerializerGenerationManager.cs

```diff
@@ -121,6 +116,13 @@ internal bool RecordTypeToGenerate(Type t, Module module, Assembly targetAssembl
             return true;
         }
 
+        private static bool IsFieldInaccessibleForSerialization(Module module, Assembly targetAssembly, FieldInfo field)
+        {
+            return field.GetCustomAttribute<NonSerializedAttribute>() == null
+                   && !SerializationManager.HasSerializer(field.FieldType)
```

Comment:
Good question.

* For compile-time codegen, we explicitly call `RegisterBuiltInSerializers` before generating code, so those serializers will be included.
* This will only ever bring in additional serializers, so we don't need to worry that it will result in *less* serializers than what we previously had.
* For run-time assembly, codegen runs *before* `FindSerializationInfo`, so serializers in the same assembly will not be included here. That's a deficiency.

We could run `FindSerializationInfo` before running codegen, that way the serializers are loaded before we rescan the assembly and generate types which don't have serializers yet.